actual/database.py

```diff
@@ -175,6 +178,17 @@ class Accounts(BaseModel, table=True):
     pending_transactions: List["PendingTransactions"] = Relationship(back_populates="account")
     transactions: List["Transactions"] = Relationship(back_populates="account")
 
+    @property
+    def balance(self) -> decimal.Decimal:
+        value = object_session(self).scalar(
+            select(func.sum(Transactions.amount)).where(
+                Transactions.acct == self.id,
+                Transactions.is_parent == 0,
+                Transactions.tombstone == 0,
+            )
+        )
+        return decimal.Decimal(value) / 100
```

Comment:
There may be a better way for this check, but it handles edge cases where accounts have no transactions:

```python
if not value:
    value = 0
return decimal.Decimal(value) / 100
```

Example account `------------------` edge case (I use them as dividers between bank accounts and credit card accounts):

```shell
balance_current=None offbudget=0 type=None closed=0 account_sync_source=None id='aaf67f5c-7fe5-460b-9d65-65668a36995b' balance_available=None balance_limit=None tombstone=0 mask=None name='------------------' official_name=None account_id=None subtype=None sort_order=32768.0 bank=None
```

Reply:
Yes, I did the coalesce on the sum so that it always has to return a value, thanks for the input.